// Copyright 2026 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package bap.common;

import "buf/validate/validate.proto";

// Defines the configuration for the workload executor action.
message WorkloadAction {
  // REQUIRED: Specifies the action to be used to execute the workload (e.g., my_repo/hlo_workload_executor@v4).
  string action = 1 [(buf.validate.field).string.min_len = 1];

  // Optional: Generic map of inputs to the action.
  map<string, string> action_inputs = 2;
}
